__tests__/commands/install/integration.js

```diff
@@ -727,13 +713,45 @@ test.concurrent('install will not overwrite files in symlinked scoped directorie
   });
 });
 
-test.concurrent.skip('install incompatible optional dependency should still install shared child dependencies',
+test.concurrent('install a module with incompatible optional dependency should skip dependency',
+  (): Promise<void> => {
+    return runInstall({}, 'install-should-skip-incompatible-optional-dep', async (config) => {
+      assert.ok(!(await fs.exists(path.join(config.cwd, 'node_modules', 'dep-incompatible'))));
+    });
+  });
+
+test.concurrent('install a module with incompatible optional dependency should skip transient dependencies',
+  (): Promise<void> => {
+    return runInstall({}, 'install-should-skip-incompatible-optional-dep', async (config) => {
+      assert.ok(!(await fs.exists(path.join(config.cwd, 'node_modules', 'dep-a'))));
+    });
+  });
+
+if (process.platform !== 'darwin') {
+  test.concurrent('install incompatible optional dependency should still install shared child dependencies',
+    (): Promise<void> => {
+      // this tests for a problem occuring due to optional dependency incompatible with os, in this case fsevents
+      // this would fail on os's incompatible with fsevents, which is everything except osx.
```

Comment:
Can you move this comment to above the if? It makes more sense there to figure out why we skip the test there.